include/fmt/core.h

```diff
@@ -82,11 +82,17 @@
 #  define FMT_HAS_CPP_ATTRIBUTE(x) 0
 #endif
 
+#ifdef _MSVC_LANG
+#  define FMT_CPLUSPLUS _MSVC_LANG
+#else
+#  define FMT_CPLUSPLUS __cplusplus
+#endif
```

Comment:
What is going on here? Doesn't MSVC use the standard `__cplusplus` macro?

Reply:
[/Zc:__cplusplus](https://docs.microsoft.com/en-us/cpp/build/reference/zc-cplusplus?view=msvc-170) enables updated C++ macro. By default `__cplusplus` is defined to some ancient value.

As I read in MSVC STL chat, the reason for this default behavior is the compatibility with some open-source code, in which higher `__cplusplus` values enable not only legitimate C++ features, but also gcc extensions.

You already had `_MSVC_LANG` for `[[fallthrough]]`, I'm just propagating it a bit further.